### ISSUE TITLE
connection-pool

### ISSUE THREAD

**Issue by chenjia123** [yidongnan/grpc-spring-boot-starter#335]
**The problem**

我注意到在GitHub上已经有人开发出了[grpc连接池](https://github.com/shimingyah/pool)，但是用golang实现的，所以我想能否在这个项目中添加此功能。

**The solution**

刚才那个项目的作者在知乎上给出了[设计思路](https://zhuanlan.zhihu.com/p/104887882#)，我想对实现应该是有帮助的，聊胜于无吧。

**Comment by ST-DDT** [comment]
IMO this should rather be added to grpc-java itself because then it has access to its internals, which would probably be useful for such an approach. See also https://github.com/grpc/grpc-java/issues/2370 for the upstream issue.

I also agree with the upstream response to this suggestion, as it is very vague in its requirements.
Which problem do you want to solve? Is it too slow? Issues with the firewall? ...
I also have the feeling that somewhere in the grpc-netty stack there is already a connection pool or at least something similar. Grpc-Javas connection creation is fast and I didn't encounter any noticeable delay in the requests.

In order to maximize the use of this "assumed" connection-pool (netty `ManagedChannel`), this library only ever creates a single `ManagedChannel` per client name, so all stubs regardless of their type will use the same underlying `ManagedChannel`.

If you really need more control over the connections e.g. then you could implement/use one of grpc-java's loadbalancers and configure them on a per client basis. This library has support for that already.

Finally, the efficient implementation of a connection pool is very tricky, so adding the pool upstream would help to find errors earlier.

TLDR: This should be handled upstream.
Pinging @yidongnan for his opinion as I might have misunderstood the question/suggestion.

**Comment by chenjia123** [comment]
Now I understand it, thank you 👍
